src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -24,6 +27,10 @@ public Task Start(IMessageSession session)
                 var task = startable.Start(session).ThrowIfNull();
 
                 var startable1 = startable;
+                var startableName = startable1.GetType().AssemblyQualifiedName;
+
+                var longRunningTokenSource = 
+                    LongRunningWarning($"The start method for {startableName} is taking a long time. This blocks endpoint startup.");
```

Comment:
"taking too long to complete", what if the user..... wants it to take that long....? :)